providers/default.rb

```diff
@@ -0,0 +1,158 @@
+#
+# Cookbook Name:: python-webapp
+# Provider:: default
+#
+# Copyright 2015, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+action :install do
+  # Use the git recipe to install git
+  run_context.include_recipe 'git'
+  run_context.include_recipe 'python::pip'
+  run_context.include_recipe 'python::virtualenv'
+
+  if new_resource.create_user
+    group new_resource.group do
+      action :create
+    end
+    user new_resource.owner do
+      action :create
+      gid new_resource.group
+    end
+  end
+
+  # If the new_resource.path is nil set the install path to
+  # `/opt/name_attribute`
+  path = new_resource.path || "/opt/#{ new_resource.name }"
+
+  if new_resource.virtualenv_path.nil?
+    virtualenv_path = "/opt/venv_#{ new_resource.name }"
+  else
+    virtualenv_path = new_resource.virtualenv_path
+  end
+
+  # If the config_destination is nil, as by default, default to
+  # /opt/<path>/settings.py
+  if new_resource.config_destination.nil?
+    config_destination = "#{ path }/settings.py"
+  else
+    config_destination = new_resource.config_destination
+  end
+
+  directory path do
```

Comment:
It passes tests without exploding? I'm not sure what sort of a mistake you're referring to.